scripts/generate_skeleton/generate_skeleton.py

```diff
@@ -0,0 +1,638 @@
+
+
+def create_tests_only(skeleton_type: str, category: str, name: str):
+    """Create test files for an existing skeleton.
+
+    Args:
+        skeleton_type: Type of skeleton ('component' or 'pipeline')
+        category: Category name (e.g., 'data_processing', 'training')
+        name: Skeleton name (e.g., 'my_processor')
+
+    Returns:
+        Path to created tests directory
+
+    Raises:
+        ValueError: If the skeleton directory or required files don't exist
+    """
+    skeleton_dir = Path(f"{skeleton_type}s/{category}/{name}")
+    main_file = skeleton_dir / f"{skeleton_type}.py"
+
+    # Check if skeleton directory exists
+    if not skeleton_dir.exists():
+        raise ValueError(
+            f"""
+Error: {skeleton_type.title()} '{name}' does not exist in category '{category}'.
+
+Expected directory: {skeleton_dir}
+
+To create this {skeleton_type} first, run:
+  python scripts/generate_skeleton/generate_skeleton.py --type={skeleton_type} --category={category} --name={name}
+
+Or use the Makefile:
+  make {skeleton_type} CATEGORY={category} NAME={name}
+""".strip()
+        )
+
+    # Check if the main skeleton file exists
+    if not main_file.exists():
+        raise ValueError(
+            f"""
+Error: {skeleton_type.title()} '{name}' directory exists but missing main file.
+
+Expected file: {main_file}
+
+The {skeleton_type} directory exists but appears incomplete. Please recreate it:
+  python scripts/generate_skeleton/generate_skeleton.py --type={skeleton_type} --category={category} --name={name}
+
+Or use the Makefile:
+  make {skeleton_type} CATEGORY={category} NAME={name}
+""".strip()
+        )
+
+    tests_dir = skeleton_dir / "tests"
+    tests_dir.mkdir(exist_ok=True)
+
+    # Generate and write test files
+    test_files = _generate_test_files(skeleton_type, category, name)
+    for filename, content in test_files.items():
+        (tests_dir / filename).write_text(content)
+
+    return tests_dir
+
+
+def main():
+    """Main entry point for the script."""
+    # Change to project root directory (script is in scripts/generate_skeleton/)
+    script_dir = Path(__file__).parent
+    project_root = script_dir.parent.parent
+    os.chdir(project_root)
+
+    parser = argparse.ArgumentParser(
+        description="Generate skeleton component or pipeline structure",
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        epilog="""
+Examples:
+  %(prog)s --type=component --category=data_processing --name=my_processor
+  %(prog)s --type=pipeline --category=ml_workflows --name=training_pipeline
+  %(prog)s --type=component --category=training --name=bert_trainer
+        """,
+    )
+
+    parser.add_argument("--type", choices=["component", "pipeline"], required=True, help="Type of skeleton to generate")
+
+    parser.add_argument(
+        "--category",
+        required=True,
+        help="Category for the component/pipeline (e.g., 'data_processing', 'training', 'ml_workflows')",
+    )
```

Comment:
The generator accepts arbitrary `--category`, but `pyproject.toml` packages only a fixed set of category modules. New categories may not be importable when installed. Suggest restricting `--category` to supported categories (existing dirs).